tests/unit/meshTests.cpp

```diff
@@ -0,0 +1,120 @@
+#define CATCH_CONFIG_MAIN
+#include "catch/catch.hpp"
+
+#include <iostream>
+#include "typedMesh.h"
+
+struct Vertex {
+    float a;
+    float b;
+    short c;
+    char d;
+};
+
+std::shared_ptr<VertexLayout> layout = std::shared_ptr<VertexLayout>(new VertexLayout({
+    {"ab", 2, GL_FLOAT, false, 0},
+    {"c",  1, GL_SHORT, false, 0},
+    {"d",  1, GL_BYTE,  false, 0},
+}));
+
+std::shared_ptr<TypedMesh<Vertex>> newMesh(unsigned int size) {
+    auto mesh = std::shared_ptr<TypedMesh<Vertex>>(new TypedMesh<Vertex>(layout, GL_TRIANGLES));
+    std::vector<Vertex> vertices;
+    for (int i = 0; i < size; ++i) {
+        vertices.push_back({0,0,0,0});
+    }
+    mesh->addVertices(std::move(vertices), {});
+    mesh->compileVertexBuffer();
+    return mesh;
+}
+
+TEST_CASE( "Simple update on vertices", "[Core][TypedMesh]" ) {
+    auto mesh = newMesh(10);
+
+    REQUIRE(mesh->getDirtyOffset() == 0);
+    REQUIRE(mesh->getDirtySize() == 0);
+
+    mesh->updateVertices(0, 4, Vertex());
+
+    REQUIRE(mesh->getDirtyOffset() == 0);
+    REQUIRE(mesh->getDirtySize() == 4 * sizeof(Vertex));
+}
+
+TEST_CASE( "Left merge on vertices with bigger left size", "[Core][TypedMesh]" ) {
+    auto mesh = newMesh(10);
+
+    mesh->updateVertices(2 * sizeof(Vertex), 2, Vertex());
+    mesh->updateVertices(0 * sizeof(Vertex), 8, Vertex());
+
+    REQUIRE(mesh->getDirtyOffset() == 0);
+    REQUIRE(mesh->getDirtySize() == 8 * sizeof(Vertex));
+}
+
+TEST_CASE( "Left merge on vertices with smaller left size", "[Core][TypedMesh]" ) {
+    auto mesh = newMesh(10);
+
+    mesh->updateVertices(2 * sizeof(Vertex), 2, Vertex());
+    mesh->updateVertices(0 * sizeof(Vertex), 1, Vertex());
+
+    REQUIRE(mesh->getDirtyOffset() == 0);
+    REQUIRE(mesh->getDirtySize() == 4 * sizeof(Vertex));
+}
+
+TEST_CASE( "Right merge on vertices dirtiness", "[Core][TypedMesh]" ) {
+    auto mesh = newMesh(10);
+
+    mesh->updateVertices(2 * sizeof(Vertex), 2, Vertex());
+    mesh->updateVertices(4 * sizeof(Vertex), 2, Vertex());
+
+    REQUIRE(mesh->getDirtyOffset() == 2 * sizeof(Vertex));
+    REQUIRE(mesh->getDirtySize() == 4 * sizeof(Vertex));
+}
+
+TEST_CASE( "Update on second attribute of the mesh for n vertices", "[Core][TypedMesh]" ) {
+    auto mesh = newMesh(10);
+    int nVert = 5;
+    size_t stride_a = 1 * sizeof(float); // stride of a in the struct
```

Comment:
just naming ... stride of a is actually 0. Should be named stride of b.

stride_a : 0
stride_b : sizeof(a)
stride_c : sizeof(a) + sizeof(b)
stride_d : sizeof(a) + sizeof(b) + sizeof(c)

Line: 96, 97 have the correct naming.
(refer https://github.com/tangrams/tangram-es/pull/182/files#diff-5a38a481e64efdae8ee018efbfd7b98aR96)